### PR TITLE
Cut scenes animation

### PR DESCRIPTION
**Pull request by raphael-pietrzak**
## Summary

This PR implements the `SceneAnimation` class for handling smooth and customizable animations within the project. The `SceneAnimation` class allows for setting target points, speeds, and pauses for each segment of the animation.

## Example

```python
target_points = [(100, 100), (200, 200), (300, 100), (800, 900)]
speeds = [100, 150, 200]  # Different speeds for each segment
pauses = [0, 1, 0.5, 2]  # Pauses at each point in seconds

self.cut_scene_animation = SceneAnimation(target_points, speeds, pauses)
```

## Checklist

- [x] I have tested this change locally and it works as expected.
- [x] I have made sure that the code follows the formatting and style guidelines of the project.

## Labels

type: enhancement, area: animation, game-core

### CODE REVIEW

File: data/database/database.db
Comment (fortwoone) [inline data/database/database.db:1]: Not sure this is meant to be listed in the commit

File: src/gui/scene_animation.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            self.reset_pause()
+
+
+        
+            
+
+        if self.has_more_targets():
+            self.move_towards_target(dt)
+        else:
+            self.is_finished = True
```
Comment (fortwoone) [inline src/gui/scene_animation.py:68]: ...I think this is slight overkill on the amount of blank lines between these two if statements 🤣

File: src/gui/scene_animation.py
Comment (fortwoone) [inline src/gui/scene_animation.py:1]: In general, this file could use more typehinting

File: src/screens/level.py [hunk 2]
```diff
@@ -52,6 +53,12 @@ def __init__(
         self.display_surface = pygame.display.get_surface()
         self.switch_screen = switch
 
+        # cut scene
+        target_points = [(100, 100), (200, 200), (300, 100), (800, 900)]
+        speeds = [100, 150, 200]  # Different speeds for each segment
+        pauses = [0, 1, 0.5, 2]  # Pauses at each point in seconds
+        self.cut_scene_animation = SceneAnimation(target_points, speeds, pauses)
```
Comment (fortwoone) [inline src/screens/level.py:60]: This is more of a wording issue. 

A **cut scene** is a scene that was removed from a movie for various reasons, and is not in the final version.
A **cutscene**, on the other hand, is what you're trying to make possible, i.e. non-interactive sequences in a game

**Review by fortwoone — COMMENTED** [review]
Some nitpicks here and there, once they have been discussed I'll give my go

File: src/gui/scene_animation.py (hunk 1 above)
Comment (raphael-pietrzak) [reply src/gui/scene_animation.py:68]: 😂😂 true

File: src/screens/level.py (hunk 2 above)
Comment (raphael-pietrzak) [reply src/screens/level.py:60]: I read your explaination on discord but i always invert 😅🙈

File: data/database/database.db
Comment (raphael-pietrzak) [reply data/database/database.db:1]: True didn't seen

**Review by fortwoone — APPROVED** [review]

**Pull request merged**